include/circt/Dialect/ESI/ESIOps.td

```diff
@@ -124,3 +130,23 @@ def NullSourceOp : ESI_Physical_Op<"null", [NoSideEffect]> {
 
   let assemblyFormat = [{ attr-dict `:` qualified(type($out)) }];
 }
+
+def NoneSourceOp : ESI_Physical_Op<"none", [NoSideEffect]> {
```

Comment:
Is there no op in the builtin or std dialects which can create a `none` constant? If not, this seems appropriate.

Reply:
Not as far as i can tell. The closest would be `handshake.source` but i don't think it's appropriate to use that in ESI. In general, it's probably only a few dialects that require none-typed constants, since their semantics are _very_ abstraction specific.